src/Slider.js

```diff
@@ -3,17 +3,49 @@ import PropTypes from 'prop-types';
 import cx from 'classnames';
 
 class Slider extends Component {
-  componentDidMount() {
-    const { fullscreen, indicators, interval, transition } = this.props;
+  constructor() {
+    super();
+    this.initSlider = this.initSlider.bind(this);
+  }
 
-    $('.slider').slider({
-      full_width: fullscreen,
+  initSlider() {
+    const { indicators, interval, duration, height } = this.props;
+    this.instance = M.Slider.init(this._slider, {
       indicators,
       interval,
-      transition
+      duration,
+      height
     });
   }
 
+  componentDidMount() {
+    this.initSlider();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.instance) {
+      this.instance.destroy();
+      var activeIndex = this.instance.activeIndex;
```

Comment:
Hey, my opinion :)

I would use `let` or `const` here.

And instead of using comments, I would create functions. This way the method would be smaller and more simple to ready.

What you think?

Reply:
You are absolutely right about `const` for `activeIndex`.

I guess functions are good to. However the functions would still need to be commented, to understand what is going on exactly.

Reply:
I see. =) Thanks @petergarnaes